iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/LimitViolationUtils.java

```diff
@@ -32,11 +32,11 @@ public static Branch.Overload checkTemporaryLimits(Branch branch, Branch.Side si
         return checkTemporaryLimits(branch, side, limitReduction, i, LimitType.CURRENT);
     }
 
-    public static Branch.Overload checkTemporaryLimits(Branch branch, Branch.Side side, float limitReduction, double i, LimitType type) {
+    public static Branch.Overload checkTemporaryLimits(Branch<?> branch, Branch.Side side, float limitReduction, double i, LimitType type) {
         Objects.requireNonNull(branch);
         Objects.requireNonNull(side);
 
-        LoadingLimits limits = branch.getLimits(type, side);
+        LoadingLimits limits = branch.getLimits(type, side).orElse(null);
```

Comment:
maybe something similar here? `return branch.getLimits(type, side).filter(...).map(limits to overload).orElse(null)`. Might not be so clear in the end.